eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+
+  - name: SdkRepoNameForSpecGen
+    type: string
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'AutoRest'
+  - name: TspConfigPath
+    type: string
+  - name: ReadmePath
+    type: string
+  - name: ApiVersion
```

Comment:
@raych1 do we need the api-version? or the SDK version? My expectation is that we will also generate the latest api version in the tspconfig file. I was just talking with @praveenkuttappan about this same question.

Reply:
Yes, we need it to support the release request on the non-latest API version without the need to update `tspconfig.yaml`.

Reply:
Is that a real scenario? I general SDK should only support the latest version.

Reply:
Yes, it's real scenario. We had a few services have been requesting to release a preview API version while the latest version in spec is another GA version after this preview version. Shanghai SDK team specially asked for this support.

Reply:
Based on rest api versioning guidelines teams shouldn't be releasing any older preview versions after they have released a GA version. If they do need to release another preview version, then their GA version shouldn't be in the spec in main yet. I don't even know if all the tooling would correctly support this scenario so we should probably talk further with the different stackholders about this requirement.

Reply:
I discussed this with the language teams, and supporting this would require additional work on the downstream SDK generation tool.

@lirenhe mentioned that there have been a few times where services requested the release of non-latest API versions. Should we consider pushing back on such requests as they violate the versioning guidelines?

Reply:
I would like to keep things simple for now and trying to support any version other than latest will likely cause issues and more complexity in various tools.

Reply:
Please check https://github.com/Azure/sdk-release-request/issues/5758 in which service team want a specific api-version instead of the latest one.   
From https://github.com/Azure/azure-rest-api-specs/blob/main/specification/deviceregistry/DeviceRegistry.Management/main.tsp ,  the latest api-version is 2024-11-01 but they want generate SDK with 2024-09-01-preview.

Such case isn't rare in mgmt. SDK, if this pipeline will be used in spec PR only, we could use the fixed latest version, but if this step serves for other purpose, I hope it should have the flexibility to support the input api version

Reply:
@lirenhe I don't mind the flexibility here, but I worry that not all the tools will correctly handle this. @lmazuel are you aware of valid scenarios where folks want to release an SDK targeting an older version of the rest api? Will the tools correctly support this scenario?

Reply:
What each language team needs to do is take the `api-version` then pass it into 'tsp-client' by 'emitter-options', which is already supported by 'tsp-client'. 
I don't want to block this PR for the merge and we might start without this feature. Later on we can add this feature when all the language tools implement it.

Reply:
I agree this is not a blocker for this PR. It is just a discussion I want to continue to ensure we really need this.

Reply:
I was talking with Ray about this yesterday and although some users will need this functionality, it is not the most common path and on the contrary, it can create confusion if users see it available.
If a user needs such a specific flow, they can manually generate the SDKs following the official documentation.

I want to make sure we get data of the number of users that run into this specific case. With that data we can re-evaluate the exposure of this functionality